main.cpp

```diff
@@ -28,510 +28,558 @@ Thing: Car Wash
-    3) Attack value
-    4) Decay value
-    5) Release value
-3 things it can do:
-    1) change oscillator cutoff
-    2) change oscillator resonance
-    3) receive host automation
- */
+int Oscillator::makeSound(int oscillatorFrequency)
+{
+    return oscillatorFrequency;
+}
+
+void Oscillator::makeOscillatorAnLFO(int oscillatorFrequency)
+{
+    oscillatorFrequency = 15;
+}
+
+int Oscillator::makeOscillatorSquareWave(int oscillatorWave)
+{
+    return oscillatorWave = 2;
+}
 
 struct Filter
 {
-//    1) Cutoff frequency
     double cutoffFrequency = 7000.4;
-//    2) Resonance
     double resonanceFrequency = 4578.93;
-//    3) Attack value
     int attack = 89;
-//    4) Decay value
     int decay = 34;
-//    5) Release value
     int release = 89;
 
-//    1) change oscillator cutoff
-    void changeCutoff(double cutoffFrequency);
-//    2) change oscillator resonance
-    void changeResonance(double resonanceFrequency);
-//    3) receive host automation
-    void getAutomationFromHost(int parameterName);
+    double changeCutoff(double cutoffFrequency);
+    double changeResonance(double resonanceFrequency);
+    int getAutomationFromHost(int parameterNumber);
 };
 
-/*
-Thing 8) Amplitude Envelope
-5 properties:
-    1) Attack value
-    2) Decay value
-    3) Sustain value
-    4) Release value
-    5) Hold value
-3 things it can do:
-    1) adjust oscillator attack
-    2) receive host automation
-    3) change dry/wet mix value
- */
+double Filter::changeCutoff(double cutoff)
+{
+    return cutoff;   
+}
+
+double Filter::changeResonance(double resonance)
+{
+    return resonance;
+}
+
+int Filter::getAutomationFromHost(int parameterNumber)
+{
+    return parameterNumber;
+}
 
 struct AmplitudeEnvelope
 {
-//    1) Attack value
     int attack = 34;
-//    2) Decay value
     int decay = 45;
-//    3) Sustain value
     int sustain = 100;
-//    4) Release value
     int release = 20;
-//    5) Hold value
     int hold = 0;
 
-//    1) adjust oscillator attack
-    void changeAttack(int attack);
-//    2) receive host automation
-    void getAutomationFromHost(int parameterName);
-//    3) change dry/wet mix value
-    void changeDryWetMix(int dryValue, int wetValue);
-
+    void changeAttack(int newAttack);
+    int getAutomationFromHost(int parameterNumber);
+    int changeDryWetMix(int wetValue);
 };
 
-/*
-Thing 9) Arpeggiator
-5 properties:
-    1) Octave range value
-    2) Number of note repetitions before moving to next note
-    3) sequence pattern number
-    4) note playing direction
-    5) pitch duration selection
-3 things it can do:
-    1) be bypassed
-    2) play held notes
-    3) play a certain note duration
- */
+void AmplitudeEnvelope::changeAttack(int newAttack)
+{
+    attack = newAttack;
+}
+
+int AmplitudeEnvelope::getAutomationFromHost(int parameterNumber)
+{
+    return parameterNumber;
+}
+
+int AmplitudeEnvelope::changeDryWetMix(int wetValue)
+{
+    int dryValue = 1;
+    return dryValue -= wetValue; 
```

Comment:
Break the habit of writing return statements like this
separate the variable modification expression with the return expression
- [ ] revise

Reply:
Think this is done.